File: app/src/main/java/com/yatatsu/passiveviewsample/dagger/module/ThreadingModule.java
package com.yatatsu.passiveviewsample.dagger.module;

import dagger.Module;
import dagger.Provides;
import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;

/**
 * ThreadingModule
 */
@Module
public class ThreadingModule {

    @Provides
    public Scheduler provideMainThread() {
        return AndroidSchedulers.mainThread();
    }
}

File: app/src/main/java/com/yatatsu/passiveviewsample/dagger/module/ThreadingModule.kt
package com.yatatsu.passiveviewsample.dagger.module

import dagger.Module
import dagger.Provides
import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers

/**
 * ThreadingModule
 */
@Module
open class ThreadingModule {

  @Provides
  open fun provideMainThread(): Scheduler {
    return AndroidSchedulers.mainThread()
  }
}
